device/modbus_mapper/README.md

```diff
@@ -0,0 +1,147 @@
+# Modbus Mapper
+
+
+## Introduction
+
+Mapper is an application that is used to connect and control devices. This is an implementation of mapper for 
+Modbus protocol. The aim is to create an application through which users can easily operate devices using ModbusTCP/ModbusRTU protocol for communication to the KubeEdge platform. The user is required to provide the mapper with the information required to control their device through the dpl configuration file. These can be changed at runtime by updating configmap.
+
+## Running the mapper
+
+  1. Please ensure that Modbus device is connected to your edge node
+  2. Set 'modbus=true' label for the node (This label is a prerequisite for the scheduler to schedule modbus_mapper pod on the node)
+
+      ```shell
+      kubectl label nodes <name-of-node> modbus=true
+      ```
+
+  3. Build and deploy the mapper by following the steps given below.
+
+### Building the modbus mapper
+
+ ```shell
+cd $GOPATH/src/github.com/kubeedge/kubeedge/device/modbus_mapper
+make # or `make modbus_mapper`
+docker tag modbus_mapper:v1.0 <your_dockerhub_username>/modbus_mapper:v1.0
+docker push <your_dockerhub_username>/modbus_mapper:v1.0
+
+Note: Before trying to push the docker image to the remote repository please ensure that you have signed into docker from your node, if not please type the followig command to sign in
+ docker login
+ # Please enter your username and password when prompted
+```
+
+### Deploying modbus mapper application
+
+```shell
+cd $GOPATH/src/github.com/kubeedge/kubeedge/device/modbus_mapper
+
+# Please enter the following details in the modbus-dpl-configmap.yaml :-
+#    1. Replace <edge_node_name> with the name of your edge node at meta.name
+#    2. Replace <dpl-configuration> with your device protocol configuration at data.dpl
+
+kubectl create -f modbus-dpl-configmap.yaml
```

Comment:
Does the mapper not use the config map cretated by the device controller in the cloud ? if we create the device model and device instance for the modbus device , then the device controller in the cloud can create the required config map and push it to the edge node.

Reply:
change to using deviceProfile.json created by device controller.